numpy/core/src/umath/loops.c.src

```diff
@@ -540,6 +542,26 @@ NPY_NO_EXPORT NPY_GCC_OPT_3 @ATTR@ void
         BINARY_LOOP_FAST(@type@, @type@, *out = in1 @OP@ in2);
     }
 }
+
+NPY_NO_EXPORT NPY_GCC_OPT_3 @ATTR@ int
+@TYPE@_@kind@@isa@_indexed(PyArrayMethod_Context *NPY_UNUSED(context),
+                           char * const*args, npy_intp const *dimensions,
+                           npy_intp const *steps, NpyAuxData *NPY_UNUSED(func))
+{
+    char *ip1 = args[0];
+    char *indx = args[1];
+    char *value = args[2];
+    npy_intp is1 = steps[0], isindex = steps[1], isb = steps[2];
+    npy_intp n = dimensions[0];
+    npy_intp i;
+    @type@ *indexed;
+    for(i = 0; i < n; i++, indx += isindex, value += isb) {
+        indexed = (@type@ *)(ip1 + is1 * *(npy_intp *)indx);
+        indexed[0] = indexed[0] @OP@ *(@type@ *)value;
```

Comment:
Totally fine to ignore, but for me it is more mental work to see a pointer calculation and then the pointer used as an array than to see its value accessed -- as in `*indexed = *indexed @OP@ *(@type@ *)value;` -- maybe just because I come from a fortran background where array names are not also treated as pointers!

p.s.  For brevity, this could be `*indexed @OP@= *(@type@ *)value;` - but perhaps less clear.